core/src/mast/merger/mod.rs

```diff
@@ -271,95 +281,101 @@ impl MastForestMerger {
     /// Remaps a nodes' potentially contained children and decorators to their new IDs according to
     /// the given maps.
     fn remap_node(&self, forest_idx: usize, node: &MastNode) -> Result<MastNode, MastForestError> {
-        let map_decorator_id = |decorator_id: &DecoratorId| {
-            self.decorator_id_mappings[forest_idx].get(decorator_id).ok_or_else(|| {
-                MastForestError::DecoratorIdOverflow(
-                    *decorator_id,
-                    self.decorator_id_mappings[forest_idx].len(),
-                )
-            })
-        };
-        let map_decorators = |decorators: &[DecoratorId]| -> Result<Vec<_>, MastForestError> {
-            decorators.iter().map(map_decorator_id).collect()
+        self.build_node_with_remapped_children(
+            node,
+            &self.node_id_mappings[forest_idx],
+            &self.decorator_id_mappings[forest_idx],
+        )
+    }
+
+    // HELPERS
+    // ================================================================================================
+
+    /// Remaps a child node ID using the node ID map, returning the original ID if not found.
+    fn remap_child(
+        &self,
+        child_id: MastNodeId,
+        nmap: &DenseIdMap<MastNodeId, MastNodeId>,
+    ) -> MastNodeId {
+        nmap.get(child_id).expect("every node id should have an entry")
+    }
```

Comment:
Should this be a free function? I’m not sure I quite understand the function though, since I'd associate "remap" with a mutation.

Reply:
The existence of this private function is to make the point that the map changes the node IDs involved in, precisely, the mutation described a little below in `build_node_with_remapped_children`.